eth/downloader/downloader.go

```diff
@@ -1019,6 +1020,7 @@ func (d *Downloader) fetchHeaders(p *peerConnection, from uint64, pivot uint64)
 							delay = n
 						}
 						headers = headers[:n-delay]
+						ignoredHeaders = delay
```

Comment:
Probably simpler if you replace `delay` altogether with `ignoredHeader` instead of defining a new `delay` instance and then just assigning it at the end.